kroxylicious/src/main/java/io/kroxylicious/proxy/frame/DecodedFrame.java

```diff
@@ -81,7 +81,6 @@ public short apiVersion() {
     @Override
     public final int estimateEncodedSize() {
         if (headerAndBodyEncodedLength != -1) {
-            assert serializationCache != null;
```

Comment:
Can we have a comment about why it must be non-null at this point?

Reply:
As in why it would be a problem if it was `null` ? Or why we shouldn't get here without it being `null`?

Reply:
The latter I think: It should never be null by the point this method gets called